### PR TITLE
[Merged by Bors] - feat(SetTheory/Cardinal/Basic): add missing `lift` versions of `iUnion` lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
Also move some lift lemmas to be next to their non-lift counterparts.

For now this does not generalize to `ι : Sort v` as this would make a mess with `PLift`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 1]
```diff
@@ -220,6 +220,11 @@ theorem lift_id' (a : Cardinal.{max u v}) : lift.{u} a = a :=
   inductionOn a fun _ => mk_congr Equiv.ulift
 #align cardinal.lift_id' Cardinal.lift_id'
 
+/-- A cardinal lifted to a lower or equal universe equals itself. -/
+@[simp, nolint simpNF]
+theorem lift_id'' (a : Cardinal.{max v u}) : lift.{u} a = a :=
+  inductionOn a fun _ => mk_congr Equiv.ulift
```
Comment (alreadydone) [inline Mathlib/SetTheory/Cardinal/Basic.lean:226]: Does this behave any differently from `lift_id'`? I think (in Lean 4) you need to explicitly specify the universes to rewrite with either of them.

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 2]
```diff
@@ -2182,27 +2187,62 @@ theorem mk_range_eq_lift {α : Type u} {β : Type v} {f : α → β} (hf : Injec
...
+    _ = sum fun i => #(f i) := mk_sigma _
+
 theorem mk_iUnion_le {α ι : Type u} (f : ι → Set α) : #(⋃ i, f i) ≤ #ι * ⨆ i, #(f i) :=
   mk_iUnion_le_sum_mk.trans (sum_le_iSup _)
 #align cardinal.mk_Union_le Cardinal.mk_iUnion_le
 
+theorem mk_iUnion_le_lift {α : Type u} {ι : Type v} (f : ι → Set α) :
+    lift.{v} #(⋃ i, f i) ≤ lift.{u} #ι * ⨆ i, lift.{v} #(f i) := by
+  refine mk_iUnion_le_sum_mk_lift.trans <| Eq.trans_le ?_ (sum_le_iSup_lift _)
+  rw [← lift_sum, lift_id''.{v,u}]
```
Comment (alreadydone) [inline Mathlib/SetTheory/Cardinal/Basic.lean:2244]: i.e., I think this works
```suggestion
  rw [← lift_sum, lift_id'.{u,v}]
```

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 3]
```diff
@@ -2214,6 +2254,11 @@ theorem mk_biUnion_le {ι α : Type u} (A : ι → Set α) (s : Set ι) :
   apply mk_iUnion_le
 #align cardinal.mk_bUnion_le Cardinal.mk_biUnion_le
 
+theorem mk_biUnion_le_lift {α : Type u} {ι : Type v} (A : ι → Set α) (s : Set ι) :
+    lift.{v}  #(⋃ x ∈ s, A x) ≤ lift.{u} #s* ⨆ x : s, lift.{v} #(A x.1) := by
```
Comment (alreadydone) [inline Mathlib/SetTheory/Cardinal/Basic.lean:2258]:
```suggestion
    lift.{v} #(⋃ x ∈ s, A x) ≤ lift.{u} #s * ⨆ x : s, lift.{v} #(A x.1) := by
```

**Review by alreadydone — COMMENTED** [review]
LGTM otherwise

File: Mathlib/SetTheory/Cardinal/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/SetTheory/Cardinal/Basic.lean]: Good catch, I must have tricked myself into thinking I needed this. Removed.

**Review by alreadydone — APPROVED** [review]
Thanks!
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**